kani-compiler/src/codegen_cprover_gotoc/reachability.rs

```diff
@@ -439,17 +439,8 @@ pub fn collect_reachable_items<'tcx>(
 /// We don't include foreign items and items that don't have MIR.
 fn should_codegen_locally<'tcx>(tcx: TyCtxt<'tcx>, instance: &Instance<'tcx>) -> bool {
     if let Some(def_id) = instance.def.def_id_if_not_guaranteed_local_codegen() {
-        if tcx.is_foreign_item(def_id) {
-            // We cannot codegen foreign items.
-            false
-        } else {
-            // TODO: This should either be an assert or a warning.
-            // Need to compile std with --always-encode-mir first though.
-            // https://github.com/model-checking/kani/issues/1605
-            // assert!(tcx.is_mir_available(def_id), "no MIR available for {:?}", def_id);
-            //(!tcx.is_mir_available(def_id)).then(|| warn!(?def_id, "Missing MIR"));
-            tcx.is_mir_available(def_id)
-        }
+        // We cannot codegen foreign items.
+        !tcx.is_foreign_item(def_id)
```

Comment:
I think I'd prefer we left a comment about `is_mir_available` and why we're NOT using it here.